packages/happy-dom/src/location/Location.ts

```diff
@@ -11,6 +11,13 @@ export default class Location extends URL {
 		super('about:blank');
 	}
 
+	/**
+	 * Override set href.
+	 */
+	set href(value: string) {
+		super.href = new URL(value, this).href;
```

Comment:
I don't think that this fixes the issue. If you set /Foo as value, the new url will throw an error

Reply:
I this is the only solution. We need to use @ts-ignore here as the types are wrong for the URL library.